feat: add open telemetry traces

ninabarbakadze opened the pull request:
## Overview

Fixes https://github.com/celestiaorg/celestia-app/issues/5702

renaynay commented on pkg/user/tx_client.go:
```diff
@@ -254,11 +256,24 @@ func SetupTxClient(
+	span.AddEvent("txclient: broadcasted PFB", trace.WithAttributes(
+		attribute.Int("num_blobs", len(blobs)),
+		attribute.String("tx_hash", resp.TxHash),
+		attribute.Float64("elapsed_seconds", time.Since(startTime).Seconds()),
```
no times needed bc span inherently does this

renaynay commented on pkg/user/tx_client.go:
```diff
@@ -609,17 +632,29 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 			// Continue polling if the transaction is still pending
 			select {
 			case <-ctx.Done():
+				span.AddEvent("txclient: context cancelled during confirmation", trace.WithAttributes(
```
we don't need this one

renaynay commented on pkg/user/tx_client.go:
```diff
@@ -628,19 +663,41 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 				return nil, executionErr
 			}
+			// if it took more than 7 seconds to confirm, log it
+			if elapsed > 7*time.Second {
```
we don't need this for traces, just add the event itself bc we'll see the time it took for it to get confirmed eventually. We would need smth like this more for metrics but we can discuss that later.

renaynay commented on pkg/user/tx_client.go:
```diff
@@ -628,19 +663,41 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 			}
 			// Resubmit straight away in the event of eviction and keep polling until tx is committed
 			_, err := client.broadcastTx(ctx, client.conns[0], client.txTracker[txHash].txBytes, signer)
 			if err != nil {
```
let's add one more event here in case of successful resubmission

renaynay reviewed (commented):
we don't need to add attributes that are individual to the tx itself here (like tx hash, height, etc) because they don't really do much for us in the context of understanding patterns across traces.

let's stick to minimal attributes like num blobs and if an error occurs, u can interpolate it as a string in the event message itself like

`span.AddEvent(fmt.Sprintf("error happened during blah blah : %w", err))`

ninabarbakadze commented:
> we don't need to add attributes that are individual to the tx itself here (like tx hash, height, etc) because they don't really do much for us in the context of understanding patterns across traces.
> 
> let's stick to minimal attributes like num blobs and if an error occurs, u can interpolate it as a string in the event message itself like
> 
> `span.AddEvent(fmt.Sprintf("error happened during blah blah : %w", err))`

Addressed your comments. Please take another look.

rootulp reviewed (approved):
LGTM

rootulp commented on pkg/user/tx_client.go:
```diff
@@ -434,9 +455,11 @@ func (client *TxClient) submitToSingleConnection(ctx context.Context, txBytes []
 		// Retry with updated sequence
 		retryTxBytes, err := client.resignTransactionWithNewSequence(txBytes)
 		if err != nil {
+			span.RecordError(fmt.Errorf("txclient/retryBroadcastingTx: rebroadcast error: %w", err))
 			return nil, err
 		}
 
+		span.AddEvent("txclient/retryBroadcastingTx: successfully rebroadcasted tx after sequence mismatch")
```
[question] `retryBroadcastingTx` doesn't match a method call above. Did it mean to say `resignTransactionWithNewSequence`?
